src/Corax/Queries/BinaryMatch.Boolean.cs

```diff
@@ -81,8 +82,16 @@ static int FillFunc(ref BinaryMatch<TInner, TOuter> match, Span<long> matches)
 
                 // need to be ready to put both outputs to the matches
                 int length = matches.Length / 2;
+                if (length < 16)
+                {
+                    // In case we get smaller matches we still acquire a cache line. 
+                    length = 16;
+                }                
+
                 Span<long> innerMatches = longBuffer.Slice(0, length);
+                Debug.Assert(innerMatches.Length == matches.Length / 2);
```

Comment:
That will fail if matches.Length is < 32

Reply:
Got rid of all that, it was flawed by design. The only real edge case is `matches.Length == 1` which requires a different way to fill.